src/scanner.l

```diff
@@ -7561,6 +7561,16 @@ static void parseMain(yyscan_t yyscanner,
   {
     BEGIN( FindMembersPHP );
   }
+  else if ( yyextra->insideJava )
+  {
+    QCString scope="java.lang.*";
+    yyextra->current->name=removeRedundantWhiteSpace(substitute(scope.left(scope.length()-1),".","::"));
```

Comment:
Since scope is a constant this can be optimized to `yyextra->current->name="java::lang";` I think. 
I don't think a trailing `::` at the end is needed.

Reply:
A `::` at the end is not necessary, intention was to use make explicitly, by ".*` that all subpackages were used but that is not necessary.

Updated pull request